Intersect.Client.Framework/Plugins/Interfaces/IClientLifecycleHelper.cs

```diff
@@ -23,47 +23,51 @@ public LifecycleChangeStateArgs(GameStates state)
         LifecycleChangeStateArgs lifecycleChangeStateArgs
     );
 
-    public class GameUpdateArgs : EventArgs
+    public class GameUpdateArgs : TimedArgs
     {
         public GameStates State { get; }
 
         public IEntity Player { get; }
 
         public IReadOnlyDictionary<Guid, IEntity> KnownEntities { get; }
 
-        public long TimeMs { get; }
-
-        public GameUpdateArgs(GameStates state, IEntity player, Dictionary<Guid, IEntity> knownEntities, long timeMs)
+        public GameUpdateArgs(GameStates state, IEntity player, Dictionary<Guid, IEntity> knownEntities, TimeSpan deltaTime)
         {
             State = state;
             Player = player;
             KnownEntities = knownEntities;
-            TimeMs = timeMs;
+            Delta = deltaTime;
         }
     }
 
-    public class GameDrawArgs : EventArgs
+    public class GameDrawArgs : TimedArgs
     {
         public DrawStates State { get; }
 
         public IEntity Entity { get; }
 
-        public long TimeMs { get; }
-
-        public GameDrawArgs(DrawStates state, long timeMs)
+        public GameDrawArgs(DrawStates state, TimeSpan deltaTime)
         {
             State = state;
-            TimeMs = timeMs;
+            Delta = deltaTime;
         }
 
-        public GameDrawArgs(DrawStates state, IEntity entity, long timeMs)
+        public GameDrawArgs(DrawStates state, IEntity entity, TimeSpan deltaTime)
         {
             State = state;
             Entity = entity;
-            TimeMs = timeMs;
+            Delta = deltaTime;
         }
     }
 
+    public class TimedArgs : EventArgs
+    {
+        /// <summary>
+        /// Time since the last update.
+        /// </summary>
+        public TimeSpan Delta { get; protected set; }
```

Comment:
There should just be a single argument constructor for this class and the property should be an auto-property not a protected setter.

Reply:
I'm not sure I follow what that means?